src/Plugin/Deriver/ContactsEntityBlockDeriver.php

```diff
@@ -53,28 +53,53 @@ public static function create(ContainerInterface $container, $base_plugin_id) {
    */
   public function getDerivativeDefinitions($base_plugin_definition) {
     foreach ($this->entityTypeManager->getDefinitions() as $entity_type_id => $entity_type) {
+      // We are only interested in entity types that have been approved.
+      if (empty($entity_type->contacts_entity)) {
+        continue;
+      }
+
       // We are only interested in the user entity itself and entities that can
       // have owners.
       $has_owner = class_implements($entity_type->getClass(), EntityOwnerInterface::class);
       if ($entity_type_id == 'user' || $has_owner) {
         $has_forms = $entity_type->hasFormClasses();
 
-        // Basic definition.
-        $this->derivatives[$entity_type_id] = $base_plugin_definition;
-        $this->derivatives[$entity_type_id]['admin_label'] = $this->t('Contacts entity form (@label)', ['@label' => $entity_type->getLabel()]);
-
-        // The entity is required for the user or types that don't have forms.
-        $this->derivatives[$entity_type_id]['context']['entity'] = new ContextDefinition('entity:' . $entity_type_id, $entity_type->getLabel(), $entity_type_id == 'user' || !$has_forms);
-        // If this can have an owner, add the context so we can do create forms.
-        if ($has_owner && $has_forms) {
-          $this->derivatives[$entity_type_id]['context']['user'] = new ContextDefinition('entity:user', $this->t('User'), FALSE);
+        // Expand out a derivative per entity bundle.
+        if ($bundle_entity_type = $entity_type->getBundleEntityType()) {
```

Comment:
Should we assume that any bundle-able entity *should* be expanded, or should we make it another flag of some kind? While it makes sense for profiles, perhaps there are other types which it doesn't?